src/realm/object-store/object_store.cpp

```diff
@@ -114,7 +114,7 @@ std::optional<CollectionType> process_collection(const Property& property)
     else if (is_dictionary(property.type)) {
         return CollectionType::Dictionary;
     }
-    return {};
+    return std::nullopt;
```

Comment:
This is [working around a bug in GCC 9](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86465) which was fixed in 10.